src/energy_grid.F90

```diff
@@ -127,16 +125,6 @@ subroutine add_grid_points(list, energy)
     i = 1
     n = size(energy)
 
-    ! If the original list is empty, we need to allocate the first element and
-    ! store first energy point
-    if (.not. associated(list)) then
```

Comment:
Can this be removed simply because you eliminated the ``pointer`` declaration above? I'm not as familiar with Fortran, so just wanted to make sure that you indeed no longer need to initialize this list anymore.